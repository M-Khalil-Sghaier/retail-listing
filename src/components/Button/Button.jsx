import classNames from "classnames";

const Button = ({ children, onClick, variant = "raised", ...props }) => {
  const baseClasses =
    "text-base w-8 h-8 inline-flex flex-shrink-0 rounded-full items-center justify-center leading-1 transition duration-200 focus:outline-none";
  const outline =
    "bg-white text-gray-800 border border-gray-800 ring-2 ring-transparent hover:bg-gray-200 active:bg-gray-50 active:border-gray-500 active:text-secondary active:ring-transparent focus-visible:border-gray-800 focus-visible:ring-gray-400";

  const raised =
    "bg-white shadow-md text-secondary border border-transparent ring-2 ring-transparent ring-offset-1 ring-offset-transparent hover:bg-gray-200 active:bg-gray-200 active:shadow-none active:border-gray-500 active:ring-transparent focus-visible:shadow-none focus-visible:ring-gray-400 focus-visible:ring-offset-white";
  const buttonClasses = classNames(
    baseClasses,
    variant === "outline" && outline,
    variant === "raised" && raised
  );
  return (
    <button className={buttonClasses} onClick={onClick} {...props}>
      {children}
    </button>
  );
};

export default Button;
