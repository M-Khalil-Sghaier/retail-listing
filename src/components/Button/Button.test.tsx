import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  describe('Render', () => {
    it('Should render correctly', () => {
      const onClick = jest.fn();
      const { container } = render(<Button onClick={onClick}>Hello</Button>);
      expect(screen.getByText('Hello')).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });
  describe('Behavior', () => {
    it('Should trigger click callback', () => {
      const onClick = jest.fn();
      render(<Button onClick={onClick}>Hello</Button>);

      const button = screen.getByRole('button');
      fireEvent.click(button);
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
