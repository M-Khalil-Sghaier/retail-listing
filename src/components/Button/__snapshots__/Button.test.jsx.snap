// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render Should render correctly 1`] = `
<div>
  <button
    class="text-base w-8 h-8 inline-flex flex-shrink-0 rounded-full items-center justify-center leading-1 transition duration-200 focus:outline-none bg-white shadow-md text-secondary border border-transparent ring-2 ring-transparent ring-offset-1 ring-offset-transparent hover:bg-gray-200 active:bg-gray-200 active:shadow-none active:border-gray-500 active:ring-transparent focus-visible:shadow-none focus-visible:ring-gray-400 focus-visible:ring-offset-white"
  >
    Hello
  </button>
</div>
`;
